### PR TITLE
[15.0][ADD] currency_rate_update_xe: new module

### PR DESCRIPTION
**Pull request by ernestotejeda**
Cc @Tecnativa TT42478

This is a new module with the same functionality as the existing one in an older version.

ping @pedrobaeza @chienandalu

### CODE REVIEW

**Review by chienandalu — APPROVED** [review]
Works great :+1:

File: currency_rate_update_xe/README.rst [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/currency-15-0/currency-15-0-currency_rate_update_xe
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runboat-Try%20me-875A7B.png
+    :target: https://runboat.odoo-community.org/webui/builds.html?repo=OCA/currency&target_branch=15.0
+    :alt: Try me on Runboat
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+This module provides adds `XE.com <https://xe.com/>`_. currency exchange rates
```
Comment (pedrobaeza) [inline currency_rate_update_xe/README.rst]:
```suggestion
This module adds `XE.com <https://xe.com/>`_. currency exchange rates
```

File: currency_rate_update_xe/README.rst [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+**Table of contents**
+
+.. contents::
+   :local:
+
+Configuration
+=============
+
+Now you can choose the 'XE.com' service when configuring currency rates providers.
```
Comment (pedrobaeza) [inline currency_rate_update_xe/README.rst]: Please put full instructions on how to get there.

File: currency_rate_update_xe/README.rst [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    :target: https://translation.odoo-community.org/projects/currency-15-0/currency-15-0-currency_rate_update_xe
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runboat-Try%20me-875A7B.png
+    :target: https://runboat.odoo-community.org/webui/builds.html?repo=OCA/currency&target_branch=15.0
+    :alt: Try me on Runboat
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+This module provides adds `XE.com <https://xe.com/>`_. currency exchange rates
+provider.
```
Comment (pedrobaeza) [inline currency_rate_update_xe/README.rst]: Put a disclaimer saying that the rates are fetched scrapping the corresponding web page. Being a public page, it's legal according sentences like https://seointel.com/news/the-court-determines-that-data-scraping-is-legal-on-linkedin/

File: currency_rate_update_xe/models/res_currency_rate_provider_XE.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    service = fields.Selection(
+        selection_add=[("XE", "XE.com")],
+        ondelete={"XE": "set default"},
+    )
+
+    def _get_supported_currencies(self):
+        self.ensure_one()
+        if self.service != "XE":
+            return super()._get_supported_currencies()
+        return self.env["res.currency"].search([]).mapped("name")
```
Comment (pedrobaeza) [inline currency_rate_update_xe/models/res_currency_rate_provider_XE.py]: Are you sure that all the currencies are supported?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Please add the logo: https://www.xe.com/themes/xe/images/apps/bots/xe-logo.png as icon

File: currency_rate_update_xe/models/res_currency_rate_provider_XE.py [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+    service = fields.Selection(
+        selection_add=[("XE", "XE.com")],
+        ondelete={"XE": "set default"},
+    )
+
+    def _get_supported_currencies(self):
+        self.ensure_one()
+        if self.service != "XE":
+            return super()._get_supported_currencies()
+        # List of currencies obrained from: https://www.xe.com/currency/
```
Comment (pedrobaeza) [inline currency_rate_update_xe/models/res_currency_rate_provider_XE.py:25]:
```suggestion
        # List of currencies obtained from: https://www.xe.com/currency/
```

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**